README.md

```diff
@@ -235,3 +240,239 @@ encryption/decryption works using Ecto Types, <br />see:
+  17, 97, 146, 103, 115, 3, 185, 230, 137, 218, 137, 209, 111, 48, 236>>
+```
+
+The hash _value_ is identical for the given input text 
+in this case the the email address `"alex@gmail.com"`.
+
+If you use the `Fields.EmailHash.dump/1` function,
+you will see the same hash value 
+(_because the same helper function is invoked_):
+
+```elixir
+iex(5)> Fields.EmailHash.dump(email)
+{:ok,
+ <<95, 251, 251, 204, 181, 59, 239, 4, 218, 193, 35, 20, 223, 131, 219, 101, 30,
+   17, 97, 146, 103, 115, 3, 185, 230, 137, 218, 137, 209, 111, 48, 236>>}
+iex(6)> Fields.EmailHash.dump(email)
+{:ok,
+ <<95, 251, 251, 204, 181, 59, 239, 4, 218, 193, 35, 20, 223, 131, 219, 101, 30,
+   17, 97, 146, 103, 115, 3, 185, 230, 137, 218, 137, 209, 111, 48, 236>>}
+```
+
+When the `EmailHash` is stored in a database
+we can lookup an email address by hashing it
+and comparing to the list. 
+
+The best way of _visualizing_ this 
+is to convert the hash value (bitstring) 
+to `base64` so that it is _human-readable_:
+
+
+```elixir
+iex(1)> email = "alex@gmail.com"
+"alex@gmail.com"
+
+iex(2)> Fields.Helpers.hash(:sha256, email) |> :base64.encode
+"X/v7zLU77wTawSMU34PbZR4RYZJncwO55onaidFvMOw="
+
+iex(3)> Fields.Helpers.hash(:sha256, email) |> :base64.encode
+"X/v7zLU77wTawSMU34PbZR4RYZJncwO55onaidFvMOw="
+```
+
+Imagine you have a database table called `people` that has just 3 columns: `id`, `email_hash` and `email_encrypted`
+
+
+| `id`  | `email_hash`  | `email_encrypted` | 
+|:-----:| ------------- | ----------------- |
+|   1   | X/v7zLU77wTawSMU34PbZR4RYZJncwO55onaidFvMOw= | MDAwMc57Y1j0nhwOdw7EvNeUVEfYQoAr7aT6oX |
+|   2   | +zXMhia/Z2I64nul6pqoDZTVM1q2K21Pby6GtPcm9iE= | MDAwMXnS1uwGN/cZRFkQgArm2Sbj9y+hnUJIS7 |
+|   3   | maY4IxoRSOSqm6qyJDrnEN1JQssJRqRGhzwOown4DPU= | MDAwMa4v0FBko++zqfAkfisXOLosQfrDLAdPax |
+
+With this "database" table, 
+we can now _lookup_ an email address to find out their `id`:
+
+```elixir
+iex(4)> Fields.Helpers.hash(:sha256, "alice@gmail.com") |> :base64.encode
+"+zXMhia/Z2I64nul6pqoDZTVM1q2K21Pby6GtPcm9iE="
+```
+
+This matches the `email_hash` in the second row of our table,
+therefore **alice's** `id` is `2` in the database.
+
+<!--
+
+[AES.encrypt/1](https://github.com/dwyl/fields/blob/519f2e9da9c6267e9b9b5359370b21a78390d020/lib/aes.ex#L30)
+has an 
+[Initialization Vector](https://en.wikipedia.org/wiki/Initialization_vector) (**`IV`**)
+which is a random set of bytes 
+prepended to the data each time it gets encrypted.
+This increases the randomness of the **`ciphertext`**
+and thus makes it more difficult to `decrypt` 
+in the event an attacker accesses the DB.
+
+
+The `IV` is included in the `bitstring` returned by `AES.encrypt/1`
+which could be split and stored separately in a high security system.
+We are storing them together for now as we feel that having a unique key 
+stored in a Key Management System (KMS) is adequate for our needs.
+
+
+
+### How does 
+
+-->
```

Comment:
Not sure if you want to keep this part commented for now or just delete it.

Reply:
Yeah, I commented it out because I didn't know how to finish it. 📝 
Basically I stoped writing the explanation because storing the IV with the cyphertext  means the effectiveness is reduced. 💭 
We can leave it for now as a reminder that we need to clarify.